src/storage/src/source/postgres/replication.rs

```diff
@@ -260,6 +261,9 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
                 Err(err) => {
                     // If the replication stream cannot be obtained in a definite way there is
                     // nothing else to do. These errors are not retractable.
+                    // TODO: Some definite errors from the stream should put the entire source
```

Comment:
We don't want to do this because we don't want to push data into the primary source's data shard––ideally, we should just make the shard totally empty.

I think the right approach is to start returning definite errors in the health stream and ensure we don't overwrite definite errors.